feature/href.ve: update plot scripts to modify the plot names (replace "." by " _" )

BinbinZhou-NOAA opened the pull request:
## Pull Request Testing ##

- [y ] Describe testing already performed for this Pull Request:</br>

- [ y] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

- [y ] Has the code been checked to ensure that no errors occur during the execution? **[Yes or No]**

- [y ] Do these updates/additions include sufficient testing updates? **[Yes or No]**

- [ y] Please complete this pull request review by **[Fill in date]**.</br>

## Pull Request Checklist ##

- [y ] Review the source issue metadata (required labels, projects, and milestone).
- [y ] Complete the PR description above.
- [ y] Ensure the PR title matches the feature branch name.
- [y ] Check the following:
- [y ]  Instructions provided on how to run
- [y ]  Developer's name is replaced by ${user} where necessary throughout the code
- [y ]  Check that the ecf file has all the proper definitions of variables
- [y ]  Check that the jobs file has all the proper settings of COMIN and COMOUT and other input variables
- [y ]  Check to see that the output directory structure is followed
- [y ]  Be sure that you are not using MET utilities outside the METplus wrapper structure

- [y ] After submitting the PR, select **Development** issue with the original issue number.
- [y ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [y ] Close the linked issue.

BinbinZhou-NOAA committed:
updateplot scripts to modify the plot names

ShelleyMelchior-NOAA commented on modulefiles/cam/cam_plots.sh:
Please check the modulefiles/cam directory in the auth EVS develop. Be sure to be all inclusive. There are modules loaded in there that are not included in your href PR. Even if href doesn't need the same modules as other parts of cam, we need the module loading to be comprehensive across all of cam. For example, the href modulefiles in this PR are missing for plots: cray-mpich, netcdf, nco, cdo, proj, and geos.

ShelleyMelchior-NOAA commented on modulefiles/cam/cam_stats.sh:
See the previous comment related to `modulefiles/cam/cam_plots.sh`

BinbinZhou-NOAA committed:
update modulefiles/cam/cam_plots.sh

ShelleyMelchior-NOAA commented:
The updates for `modulefiles/cam_plots.sh` look good. Can you do the same for `modulefiles/cam_stats.sh`?

BinbinZhou-NOAA commented:
Shelley,

  modulefiles/cam/cam_plots.sh has been updated to same as that in
authorized repo.

Binbin


On Sat, Jul 8, 2023 at 7:14 PM Shelley Melchior ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> On modulefiles/cam/cam_stats.sh
> <https://github.com/NOAA-EMC/EVS/pull/156#discussion_r1257384724>:
>
> See the previous comment related to modulefiles/cam/cam_plots.sh
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/156#pullrequestreview-1520823678>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFKX6XUG5QYIKAMJQSTXPHSURANCNFSM6AAAAAA2DB4WQA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

BinbinZhou-NOAA committed:
update modulefiles/cam/cam_stats.sh

BinbinZhou-NOAA commented:
Sure, I have updated modulefiles/cam/cam_stats.sh

Thanks!

Binbin

On Mon, Jul 10, 2023 at 8:24 AM Shelley Melchior ***@***.***>
wrote:

> The updates for modulefiles/cam_plots.sh look good. Can you do the same
> for modulefiles/cam_stats.sh?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/156#issuecomment-1628854809>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFN775KIU7O5KOXFZ3LXPPYAJANCNFSM6AAAAAA2DB4WQA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

LoganDawson-NOAA commented on scripts/cam/plots/exevs_href_spcoutlook_plots.sh:
```diff
@@ -179,10 +179,10 @@ for domain in day1_mrgl day1_slgt day1_tstm day1_enh day1_mdt day1_high day2_mrg
-  mv threshold_average_regional_${domain}_${valid}_${var}_fbias*.png  evs.href.fbias.${var_new}_${level}.last${past_days}days.thresholdmean.${valid}.buk_${domain}.png
-  mv performance_diagram_regional_${domain}_${valid}_${var}*.png evs.href.ctc.${var_new}_${level}.last${past_days}days.perfdiag.${valid}.buk_${domain}.png
+  mv lead_average_regional_${domain}_valid_available_times_${var}*.png  evs.href.csi_fbias.${var_new}_${level}.last${past_days}days.fhrmean_${valid}.buk_${domain}.png
+  mv threshold_average_regional_${domain}_${valid}_${var}_csi*.png  evs.href.csi.${var_new}_${level}.last${past_days}days.thresholdmean_${valid}.buk_${domain}.png
```
On lines 183 and 184 here, "thresholdmean" should be shortened to "threshmean"

LoganDawson-NOAA commented on scripts/cam/plots/exevs_href_precip_spatial_plots.sh:
```diff
@@ -72,12 +72,11 @@ for model in $model_list ; do
     past=`$NDATE -$fhr ${VDATE}12`
     INITDATE=${past:0:8}    
-    mv ${model}.v${VDATE}12.0${fhr}h.${mask}.png  evs.${model}.spatial_map.apcp_a24.init${INITDATE}.vlid${VDATE}12.f${fhr}.buk_${mask}.png
+    mv ${model}.v${VDATE}12.0${fhr}h.${mask}.png  evs.${model}.spatial_map.apcp_a24.vlid12z.f${fhr}.buk_${mask}.png
```
for spatial maps, we're looking for filenames that match: ${model}.v${VDATE}12.${fhr}.${mask}.gif

$fhr should be a 3-digit value (024, 036, 048)
$mask should just be "conus" and not "buk_conus"

These updates will help streamline Mallory's development of the QPF vs. QPE webpage

BinbinZhou-NOAA commented:
Logan,

  Is change  also for other plot scripts, not only for
exevs_href_spcoutlook_plots.sh?

Binbin


On Mon, Jul 10, 2023 at 10:50 AM Logan Dawson ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In scripts/cam/plots/exevs_href_spcoutlook_plots.sh
> <https://github.com/NOAA-EMC/EVS/pull/156#discussion_r1258385019>:
>
> > @@ -179,10 +179,10 @@ for domain in day1_mrgl day1_slgt day1_tstm day1_enh day1_mdt day1_high day2_mrg
>      level=ml
>      valid=valid_00z_12z
>    fi
> -  mv lead_average_regional_${domain}_valid_available_times_${var}*.png  evs.href.csi_fbias.${var_new}_${level}.last${past_days}days.fhrmean.${valid}.buk_${domain}.png
> -  mv threshold_average_regional_${domain}_${valid}_${var}_csi*.png  evs.href.csi.${var_new}_${level}.last${past_days}days.thresholdmean.${valid}.buk_${domain}.png
> -  mv threshold_average_regional_${domain}_${valid}_${var}_fbias*.png  evs.href.fbias.${var_new}_${level}.last${past_days}days.thresholdmean.${valid}.buk_${domain}.png
> -  mv performance_diagram_regional_${domain}_${valid}_${var}*.png evs.href.ctc.${var_new}_${level}.last${past_days}days.perfdiag.${valid}.buk_${domain}.png
> +  mv lead_average_regional_${domain}_valid_available_times_${var}*.png  evs.href.csi_fbias.${var_new}_${level}.last${past_days}days.fhrmean_${valid}.buk_${domain}.png
> +  mv threshold_average_regional_${domain}_${valid}_${var}_csi*.png  evs.href.csi.${var_new}_${level}.last${past_days}days.thresholdmean_${valid}.buk_${domain}.png
>
> On lines 183 and 184 here, "thresholdmean" should be shortened to
> "threshmean"
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/156#pullrequestreview-1522274377>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFLOH4WNYPMMGBOR5N3XPQJCFANCNFSM6AAAAAA2DB4WQA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

BinbinZhou-NOAA committed:
update plot scripts for graphic file name change

BinbinZhou-NOAA commented:
All href.ve plot scripts have been updated for Logan's suggestions.

Binbin

On Mon, Jul 10, 2023 at 10:55 AM Logan Dawson ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In scripts/cam/plots/exevs_href_precip_spatial_plots.sh
> <https://github.com/NOAA-EMC/EVS/pull/156#discussion_r1258392649>:
>
> > @@ -72,12 +72,11 @@ for model in $model_list ; do
>    for fhr in 24 48 ; do
>      past=`$NDATE -$fhr ${VDATE}12`
>      INITDATE=${past:0:8}
> -    mv ${model}.v${VDATE}12.0${fhr}h.${mask}.png  evs.${model}.spatial_map.apcp_a24.init${INITDATE}.vlid${VDATE}12.f${fhr}.buk_${mask}.png
> +    mv ${model}.v${VDATE}12.0${fhr}h.${mask}.png  evs.${model}.spatial_map.apcp_a24.vlid12z.f${fhr}.buk_${mask}.png
>
> for spatial maps, we're looking for filenames that match:
> ${model}.v${VDATE}12.${fhr}.${mask}.gif
>
> $fhr should be a 3-digit value (024, 036, 048)
> $mask should just be "conus" and not "buk_conus"
>
> These updates will help streamline Mallory's development of the QPF vs.
> QPE webpage
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/156#pullrequestreview-1522284656>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFPNFUUL77IDGJG7EPDXPQJT5ANCNFSM6AAAAAA2DB4WQA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

LoganDawson-NOAA commented:
@BinbinZhou-NOAA yes, the change from "thresholdmean" to "threshmean" applies for all plots, not just the SPC outlook plots. Thanks for making those updates

PerryShafran-NOAA commented:
@LoganDawson-NOAA - do we need to remove the "buk" from all plots as well?  Apologies for using Binbin's PR to ask this, but I think I may still have those in there.

LoganDawson-NOAA commented:
@PerryShafran-NOAA the comment about changing "buk_conus" to "conus" was specific to the 24-h QPF spatial maps that will be used on [this webpage](https://www.emc.ncep.noaa.gov/users/verification/precip/verif/daily/). The spatial maps have a different naming convention than other verification graphics. The vast majority of the EVS graphics for CONUS **_should_** have "buk_conus" in the filename.

PerryShafran-NOAA commented:
OK, thanks for that clarification.  

Perry

BinbinZhou-NOAA commented:
Logan,

   global_ens and sref also have spatial_map. The "buk"  should be removed
from their spatial map file names as well?

Binbin

On Mon, Jul 10, 2023 at 11:36 AM Logan Dawson ***@***.***>
wrote:

> @PerryShafran-NOAA <https://github.com/PerryShafran-NOAA> the comment
> about changing "buk_conus" to "conus" was specific to the 24-h QPF spatial
> maps that will be used on this webpage
> <https://www.emc.ncep.noaa.gov/users/verification/precip/verif/daily/>.
> The spatial maps have a different naming convention than other verification
> graphics. The vast majority of the EVS graphics for CONUS *should* have
> "buk_conus" in the filename.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/156#issuecomment-1629210309>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFNFPLFQSBDIXQEYJ6TXPQOQDANCNFSM6AAAAAA2DB4WQA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

LoganDawson-NOAA commented:
@BinbinZhou-NOAA yes. Since the QPF vs. QPE maps will be using images from many different EVS components, all of the filenames need to follow the same convention to ease the development and functionality of the new webpage

ShelleyMelchior-NOAA commented:
@BinbinZhou-NOAA -- I ran precip and grid2obs stats for href. Please review.
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/cam`
logfiles:
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_href_precip_stat.o81695162`
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_href_g2o_stat.o81695161`

Once approved I will start the plotting processes.

BinbinZhou-NOAA commented:
Shelley,

  It seems both  precip and grid2obs stat of href.ve testing runs  were not
completed yesterday.  Actually my runs were not completed
yesterday either. I guess it was due to the outage of dogwood yesterday. My
precip and grid2obs runs were completed today.
So could you please re-run both precip and grid2obs today?

Thanks!

Binbin

On Mon, Jul 10, 2023 at 7:46 PM Shelley Melchior ***@***.***>
wrote:

> @BinbinZhou-NOAA <https://github.com/BinbinZhou-NOAA> -- I ran precip and
> grid2obs stats for href. Please review.
> COMOUT=/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/cam
> logfiles:
> /lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_href_precip_stat.o81695162
> /lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_href_g2o_stat.o81695161
>
> Once approved I will start the plotting processes.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/156#issuecomment-1629882146>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFNC7IVAZ5TU4Q4QJI3XPSH3TANCNFSM6AAAAAA2DB4WQA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

ShelleyMelchior-NOAA commented:
@BinbinZhou-NOAA -- I re-ran precip and grid2obs stats for href. Please review.
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/cam`
logfiles:
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_href_precip_stat.o81815845`
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_href_g2o_stat.o81815843`

Once approved I will start the plotting processes.

BinbinZhou-NOAA commented:
Shelley,

    Both precip and grid2obs output stat files  are in good shape.

Binbin

On Tue, Jul 11, 2023 at 2:19 PM Shelley Melchior ***@***.***>
wrote:

> @BinbinZhou-NOAA <https://github.com/BinbinZhou-NOAA> -- I re-ran precip
> and grid2obs stats for href. Please review.
> COMOUT=/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/cam
> logfiles:
> /lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_href_precip_stat.o81815845
> /lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_href_g2o_stat.o81815843
>
> Once approved I will start the plotting processes.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/156#issuecomment-1631283035>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFLM6GW2SH5HKIBSMDLXPWKMLANCNFSM6AAAAAA2DB4WQA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

ShelleyMelchior-NOAA commented:
@BinbinZhou-NOAA , I will run plots next. Can you please update your feature branch now that Perry has made a few PR merges?

BinbinZhou-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/href.ve

BinbinZhou-NOAA commented:
Shelley,

I have synced the feature/href.ve with the authorized repo.

Binbin



On Tue, Jul 11, 2023 at 4:07 PM Shelley Melchior ***@***.***>
wrote:

> @BinbinZhou-NOAA <https://github.com/BinbinZhou-NOAA> , I will run plots
> next. Can you please update your feature branch now that Perry has made a
> few PR merges?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/156#issuecomment-1631442150>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFMIB36SVC7OBJL6Z4DXPWW77ANCNFSM6AAAAAA2DB4WQA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

ShelleyMelchior-NOAA commented:
Thanks, @BinbinZhou-NOAA . I have completed running all of the plots processing. Please have a look.
COMOUT=`/lfs/h2/emc/ptmp/shelley.melchior/evs/v1.0/plots/cam`
logfiles: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_href_*_plots.o*`
You can untar and verify the proper png graphic names.

BinbinZhou-NOAA commented:
Shelley,

    I have checked all files, all are good.
    I just got notice fro Mallory that the spatial map file names do not
match the spatial map
naming standard of EVS, so I have to modify them again. Please do not close
the href.ve PR

Thanks!

Binbin

Binbin


On Tue, Jul 11, 2023 at 5:09 PM Shelley Melchior ***@***.***>
wrote:

> Thanks, @BinbinZhou-NOAA <https://github.com/BinbinZhou-NOAA> . I have
> completed running all of the plots processing. Please have a look.
> COMOUT=/lfs/h2/emc/ptmp/shelley.melchior/evs/v1.0/plots/cam
> logfiles: /lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_href_*_plots.o*
> You can untar and verify the proper png graphic names.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/156#issuecomment-1631511845>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFM6Z3QCG6H7HVO5O7TXPW6GLANCNFSM6AAAAAA2DB4WQA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

BinbinZhou-NOAA committed:
change the spatial map file name to match naming standard

BinbinZhou-NOAA committed:
correct cpu request in spatial map job

BinbinZhou-NOAA committed:
update scripts/cam/plots/exevs_href_precip_spatial_plots.sh again

BinbinZhou-NOAA commented:
Shelley,

  The spatial map plot file names have been corrected and updated in my
href.ve fork.
Please test it again (just run the precip spatial map job).

Binbin

On Wed, Jul 12, 2023 at 10:41 AM Binbin Zhou - NOAA Affiliate <
***@***.***> wrote:

> Shelley,
>
>     I have checked all files, all are good.
>     I just got notice fro Mallory that the spatial map file names do not
> match the spatial map
> naming standard of EVS, so I have to modify them again. Please do not
> close the href.ve PR
>
> Thanks!
>
> Binbin
>
> Binbin
>
>
> On Tue, Jul 11, 2023 at 5:09 PM Shelley Melchior ***@***.***>
> wrote:
>
>> Thanks, @BinbinZhou-NOAA <https://github.com/BinbinZhou-NOAA> . I have
>> completed running all of the plots processing. Please have a look.
>> COMOUT=/lfs/h2/emc/ptmp/shelley.melchior/evs/v1.0/plots/cam
>> logfiles: /lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_href_*_plots.o*
>> You can untar and verify the proper png graphic names.
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/NOAA-EMC/EVS/pull/156#issuecomment-1631511845>, or
>> unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AQBMPFM6Z3QCG6H7HVO5O7TXPW6GLANCNFSM6AAAAAA2DB4WQA>
>> .
>> You are receiving this because you were mentioned.Message ID:
>> ***@***.***>
>>
>
>
> --
>
> Binbin Zhou
>
> Physical Scientist
>
> Lynker at NOAA/NWS/NCEP/EMC
>
> 5830 University Research Ct.
>
> College Park, MD 20740
>
> ***@***.***
>
> 301-683-3683
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

ShelleyMelchior-NOAA commented:
I see the updates. Thanks @BinbinZhou-NOAA . 

Before I make the repeat test, there is one more change that we need to address to meet EE2 compliance as part of the internal code review by EIB. The issue to address is the definition of COMIN and COMOUT variables lower than `jobs/`. 

So `scripts/cam/plots/exevs_href_precip_spatial_plots.sh` needs to be updated to bring the COMROOT variable definition out of that file and up into `JEVS_CAM_PLOTS`.

BinbinZhou-NOAA committed:
relocate COMROOT definition in spatial map job

BinbinZhou-NOAA commented:
Shelley,

   I just moved the COMROOT to the plot JOB file and updated it in the
href.ve fork on github.

Binbin

On Wed, Jul 12, 2023 at 1:47 PM Shelley Melchior ***@***.***>
wrote:

> I see the updates. Thanks @BinbinZhou-NOAA
> <https://github.com/BinbinZhou-NOAA> .
>
> Before I make the repeat test, there is one more change that we need to
> address to meet EE2 compliance as part of the internal code review by EIB.
> The issue to address is the definition of COMIN and COMOUT variables lower
> than jobs/.
>
> So scripts/cam/plots/exevs_href_precip_spatial_plots.sh needs to be
> updated to bring the COMROOT variable definition out of that file and up
> into JEVS_CAM_PLOTS.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/156#issuecomment-1632957760>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFKUQKF5JTIRLMV6I6LXP3PMRANCNFSM6AAAAAA2DB4WQA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

ShelleyMelchior-NOAA commented:
Good news, @BinbinZhou-NOAA  -- I finished the re-run for the precip spatial map job.
COMOUT=`/lfs/h2/emc/ptmp/shelley.melchior/evs/v1.0/plots/cam/atmos.20230711`
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_href_precip_spatial_plots.o82265639`

BinbinZhou-NOAA commented:
Shelley,

   Yes, all gif files inside the tar file are good.
Thanks!

Binbin

On Thu, Jul 13, 2023 at 2:53 PM Shelley Melchior ***@***.***>
wrote:

> Good news, @BinbinZhou-NOAA <https://github.com/BinbinZhou-NOAA> -- I
> finished the re-run for the precip spatial map job.
> COMOUT=/lfs/h2/emc/ptmp/shelley.melchior/evs/v1.0/plots/cam/atmos.20230711
> logfile:
> /lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_href_precip_spatial_plots.o82265639
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/156#issuecomment-1634735954>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFOXOV7TFLN7Z6YDGR3XQA7ZNANCNFSM6AAAAAA2DB4WQA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

ShelleyMelchior-NOAA commented:
Great! I will do the final code review next.

ShelleyMelchior-NOAA commented on jobs/cam/stats/JEVS_CAM_STATS:
@BinbinZhou-NOAA @LoganDawson-NOAA  -- since JEVS_CAM_STATS is shared across all cam, can you confirm that these changes are expected and OK for other models?

LoganDawson-NOAA replied:
@ShelleyMelchior-NOAA thanks for flagging this. It looks like this PR includes an older version of JEVS_CAM_STATS that doesn't retain changes I included in my radar/severe stats PR. My jobs will definitely break with these changes.

@BinbinZhou-NOAA please copy the version of JEVS_CAM_STATS from the develop branch and include it in this PR. Your code should still run fine because I accounted for your HREF use cases in the logic I added.

ShelleyMelchior-NOAA commented on modulefiles/v1.0:
@BinbinZhou-NOAA @PerryShafran-NOAA @LoganDawson-NOAA  -- what is this file? What is using it?

LoganDawson-NOAA replied:
@ShelleyMelchior-NOAA the `modulefiles/v1.0` file was added to the repository back in December with a global_ens PR. It's being sourced in a script that compiles code for GEFS and SREF.

I'm not sure what's the most appropriate naming convention, but I think it should probably be named more clearly so we know what its intended use is

ShelleyMelchior-NOAA replied:
Got it. Thanks. A rename is a good idea. I don't have any good ideas. For now, I suggest two changes, @BinbinZhou-NOAA :
1 - change line 5 to: `## Required to compile executables for EVS/v1.0 on WCOSS2`
2 - correct the typon on line 7 ("to" not "ot")

malloryprow replied:
I use versions/build.ver to source the build version numbers and just do the module loading in build_evs_global_det_atmos.sh. I think Binbin and I are the only one with executables that need to be built, so thought I'd share my process.

malloryprow replied:
Having versions/build.ver is inspired from the global workflow https://github.com/NOAA-EMC/global-workflow/tree/develop/versions

ShelleyMelchior-NOAA replied:
Thanks @malloryprow . So that begs the question, should we be consistent across components and have those components that require the building of executables source the same file? And if so, should we clean it up as a "day 2" thing in a different PR (or even EVSv2) so this one can get wrapped up?

BinbinZhou-NOAA commented:
Shelley, Logan,

  I am wondering why my JEVS_CAM_STATS is still  the old version.
Sure I'll copy the version from authorized repo.

Thanks!

Binbin


On Thu, Jul 13, 2023 at 8:09 PM Logan Dawson ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> On jobs/cam/stats/JEVS_CAM_STATS
> <https://github.com/NOAA-EMC/EVS/pull/156#discussion_r1263154095>:
>
> @ShelleyMelchior-NOAA <https://github.com/ShelleyMelchior-NOAA> thanks
> for flagging this. It looks like this PR includes an older version of
> JEVS_CAM_STATS that doesn't retain changes I included in my radar/severe
> stats PR. My jobs will definitely break with these changes.
>
> @BinbinZhou-NOAA <https://github.com/BinbinZhou-NOAA> please copy the
> version of JEVS_CAM_STATS from the develop branch and include it in this
> PR. Your code should still run fine because I accounted for your HREF use
> cases in the logic I added.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/156#discussion_r1263154095>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFLH2YJJOEWQE4EQKLDXQCE4NANCNFSM6AAAAAA2DB4WQA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

BinbinZhou-NOAA commented:
Yes, as Logan said, the file v1.0 file defined required modules used to
compile several Fortran
codes for  GEFS and SREF.  If you have a better name for it. I can change
the file name.

Binbin

Binbin

On Thu, Jul 13, 2023 at 8:04 PM Logan Dawson ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> On modulefiles/v1.0
> <https://github.com/NOAA-EMC/EVS/pull/156#discussion_r1263151706>:
>
> @ShelleyMelchior-NOAA <https://github.com/ShelleyMelchior-NOAA> the
> modulefiles/v1.0 file was added to the repository back in December with a
> global_ens PR. It's being sourced in a script that compiles code for GEFS
> and SREF.
>
> I'm not sure what's the most appropriate naming convention, but I think it
> should probably be named more clearly so we know what its intended use is
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/156#discussion_r1263151706>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFKOMLRZMTY4AVJ3JP3XQCEIZANCNFSM6AAAAAA2DB4WQA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

BinbinZhou-NOAA commented:
Sure
Thanks!

Binbin

On Thu, Jul 13, 2023 at 8:13 PM Shelley Melchior ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> On modulefiles/v1.0
> <https://github.com/NOAA-EMC/EVS/pull/156#discussion_r1263155470>:
>
> Got it. Thanks. A rename is a good idea. I don't have any good ideas. For
> now, I suggest two changes, @BinbinZhou-NOAA
> <https://github.com/BinbinZhou-NOAA> :
> 1 - change line 5 to: ## Required to compile executables for EVS/v1.0 on
> WCOSS2
> 2 - correct the typon on line 7 ("to" not "ot")
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/156#discussion_r1263155470>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFNPE2P3HOFL3HXJTU3XQCFJTANCNFSM6AAAAAA2DB4WQA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

BinbinZhou-NOAA commented:
Yes, I think this can be considered in day 2 .

Binbin

On Fri, Jul 14, 2023 at 8:48 AM Shelley Melchior ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> On modulefiles/v1.0
> <https://github.com/NOAA-EMC/EVS/pull/156#discussion_r1263697882>:
>
> Thanks @malloryprow <https://github.com/malloryprow> . So that begs the
> question, should we be consistent across components and have those
> components that require the building of executables source the same file?
> And if so, should we clean it up as a "day 2" thing in a different PR (or
> even EVSv2) so this one can get wrapped up?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/156#discussion_r1263697882>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFMXXQJRTTCD5NI4X5TXQE523ANCNFSM6AAAAAA2DB4WQA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

BinbinZhou-NOAA committed:
update JEVS_CAM_STATS and v1.0 file

BinbinZhou-NOAA commented:
Shelley,

  Both JEVS_CAM_STATS and v1.0 files have been updated and  sent to feature/
href.ve fork on gitHub.
The JEVS_CAM_STATS was directly copied from the authorized develop repo (no
any change).
I have tested this change, everything is fine.

Binbin

On Fri, Jul 14, 2023 at 9:26 AM Binbin Zhou - NOAA Affiliate <
***@***.***> wrote:

> Yes, I think this can be considered in day 2 .
>
> Binbin
>
> On Fri, Jul 14, 2023 at 8:48 AM Shelley Melchior ***@***.***>
> wrote:
>
>> ***@***.**** commented on this pull request.
>> ------------------------------
>>
>> On modulefiles/v1.0
>> <https://github.com/NOAA-EMC/EVS/pull/156#discussion_r1263697882>:
>>
>> Thanks @malloryprow <https://github.com/malloryprow> . So that begs the
>> question, should we be consistent across components and have those
>> components that require the building of executables source the same file?
>> And if so, should we clean it up as a "day 2" thing in a different PR (or
>> even EVSv2) so this one can get wrapped up?
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/NOAA-EMC/EVS/pull/156#discussion_r1263697882>, or
>> unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AQBMPFMXXQJRTTCD5NI4X5TXQE523ANCNFSM6AAAAAA2DB4WQA>
>> .
>> You are receiving this because you were mentioned.Message ID:
>> ***@***.***>
>>
>
>
> --
>
> Binbin Zhou
>
> Physical Scientist
>
> Lynker at NOAA/NWS/NCEP/EMC
>
> 5830 University Research Ct.
>
> College Park, MD 20740
>
> ***@***.***
>
> 301-683-3683
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

ShelleyMelchior-NOAA commented:
Great! @BinbinZhou-NOAA , I think we are in a good place w/ this PR now. I have informally completed the code review. I will approve and merge.

ShelleyMelchior-NOAA reviewed (approved):
Code changes are as expected. Approved.
